numerai_tools/submissions.py

```diff
@@ -16,6 +19,105 @@
 SIGNALS_ALLOWED_PRED_COLS = ["prediction", "signal"]
 
 
+def validate_headers(
+    expected_id_cols: List[str], expected_pred_cols: List[str], submission: pd.DataFrame
+) -> Tuple[str, str]:
+    """Validate the given submission has the right headers.
+
+    Arguments:
+        submission -- pandas DataFrame of the submission
+
+    Return Tuple[str, str]:
+        - string name of the id column
+        - string name of the prediction column
+    """
+    expected_headers = [
+        [ticker_col, signal_col]
+        for ticker_col in expected_id_cols
+        for signal_col in expected_pred_cols
+    ]
+    columns = submission.columns
+    valid_headers = list(columns) in expected_headers
+    assert (
+        valid_headers
+    ), f"headers must be one of {expected_id_cols} and one of {expected_pred_cols}"
+    return columns[0], columns[1]
+
+
+def validate_headers_numerai(submission: pd.DataFrame) -> Tuple[str, str]:
+    return validate_headers(
+        NUMERAI_ALLOWED_ID_COLS, NUMERAI_ALLOWED_PRED_COLS, submission
+    )
+
+
+def validate_headers_signals(submission: pd.DataFrame) -> Tuple[str, str]:
+    return validate_headers(
+        SIGNALS_ALLOWED_ID_COLS, SIGNALS_ALLOWED_PRED_COLS, submission
+    )
+
+
+def validate_values(submission: pd.DataFrame, prediction_col: str) -> None:
+    """
+    Validates the given submission's values are between 0 and 1 exclusive and
+    that the submission have a non-zero standard deviation.
+
+    Arguments:
+        submission -- pandas DataFrame of the submission
+        prediction_col -- the string name of the prediction column returned by validate_headers
+    """
+    assert (
+        submission[prediction_col].isna().sum() == 0
+    ), "submission must not contain NaNs"
+    assert (
+        submission[prediction_col].between(0, 1).all()
+    ), "values must be between 0 and 1 exclusive"
```

Comment:
This would potentially lead to a very long error message and it's trivial for a user to check/correct for this, so I think the simpler error message is best here.